solr/solrj/src/java/org/apache/solr/common/util/ExecutorUtil.java

```diff
@@ -143,6 +173,37 @@ static void awaitTermination(ExecutorService pool, long timeout, TimeUnit unit)
     }
   }
 
+  /**
+   * Await the termination of an {@link ExecutorService} until all threads are complete, or until we
+   * are interrupted, at which point the {@link ExecutorService} will be interrupted as well.
+   *
+   * @param pool the ExecutorService to wait on
+   */
+  public static void awaitTerminationForever(ExecutorService pool) {
+    boolean shutdown = false;
+    while (!shutdown) {
+      try {
+        try {
+          // Wait a while for existing tasks to terminate
+          shutdown = pool.awaitTermination(60, TimeUnit.SECONDS);
+        } catch (InterruptedException ie) {
+          // Force cancel if current thread also interrupted
+          pool.shutdownNow();
+          // Preserve interrupt status
+          Thread.currentThread().interrupt();
```

Comment:
Arguably, if we've given up waiting for awaiting termination, then the interruption has done its job and we don't need to keep the flag.